DashAI/front/src/components/experiments/PrepareDatasetStep.jsx

```diff
@@ -0,0 +1,398 @@
+
+  // rows numbers state
+  const defaultParitionsIndex = {
+    training: [],
+    validation: [],
+    testing: [],
+  };
+  const defaultPartitionsPercentage = {
+    training: 0,
+    validation: 0,
+    testing: 0,
+  };
+  const [rowsPartitionsIndex, setRowsPartitionsIndex] = useState(
+    defaultParitionsIndex,
+  );
+  const [rowsPartitionsPercentage, setRowsPartitionsPercentage] = useState(
+    defaultPartitionsPercentage,
+  );
+  const [splitsReady, setSplitsReady] = useState(false);
+
+  // handle rows numbers change state
+  const [rowsPreference, setRowsPreference] = useState("random");
+  const [rowsPartitionsError, setRowsPartitionsError] = useState(false);
+  const [rowsPartitionsErrorText, setRowsPartitionsErrorText] = useState("");
+
+  const parseRangeToIndex = (value, total) => {
+    const numbersArray = [];
+    if (!rangeRegex.test(value)) {
+      throw new Error(
+        "Ranges and indexes must match the format on the example above",
+      );
+    }
+    const ranges = value.split(",");
+    ranges.forEach((range) => {
+      const [min, max] = range.split("-");
+      if (!range.includes("-") && parseInt(range) <= total) {
+        numbersArray.push(parseInt(range));
+      } else if (
+        (!range.includes("-") && parseInt(range) > total) ||
+        parseInt(max) > total
+      ) {
+        throw new Error(
+          "The indexes should be minor than the total of columns or rows",
+        );
+      } else if (parseInt(min) > parseInt(max)) {
+        throw new Error(
+          "The second number of a range must be greater than the first",
+        );
+      } else {
+        for (let i = parseInt(min); i <= parseInt(max); i++) {
+          numbersArray.push(i);
+        }
+      }
+    });
+    return numbersArray;
+  };
+
+  const checkSplit = (training, validation, testing) => {
+    if (parseInt(training) + parseInt(validation) + parseInt(testing) === 100) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+  const handleInputColumnsChange = (event) => {
+    const input = event.target.value.replace(/ /g, ""); // TODO: dont accept spaces between numbers
+    try {
+      const columnNumbers = parseRangeToIndex(input, totalColumns);
+      setParseInputColumnsError(false);
+      setInputColumns(columnNumbers);
+    } catch (error) {
+      setParseInputColumnsErrorText(error.message);
+      setParseInputColumnsError(true);
+    }
+  };
+  const handleOutputColumnsChange = (event) => {
+    const input = event.target.value.replace(/ /g, "");
+    try {
+      const columnIndex = parseRangeToIndex(input, totalColumns); // TODO: input and output columns should be less than total
+      setParseOutputColumnsError(false);
+      setOutputColumns(columnIndex);
+    } catch (error) {
+      setParseOutputColumnsErrorText(error.message);
+      setParseOutputColumnsError(true);
+    }
+  };
+  const handleRowsPreferenceChange = (event) => {
+    if (event.target.value === "manually") {
```

Comment:
I think this would look better as `splitByIndex`.
```suggestion
    if (event.target.value === "splitByIndex") {
```

Therefore, the other option could be called `splitByPercentage.`